### PR TITLE
Fix publishing daily nupkg to MyGet

### LINKED ISSUE #7955: Daily builds not published any more in the last months to MyGet
Steps to reproduce
------------------

Run the install-powershell.ps1 script [here](https://github.com/PowerShell/PowerShell/blob/master/tools/install-powershell.ps1) with the `-Daily` switch (`.\tools\install-powershell.ps1 -Daily`) or just `iex "& { $(irm 'https://aka.ms/install-pscore' ) } -Daily"` for the faint-hearted ;-)

Expected behavior
-----------------

Downloaded PowerShell version from [MyGet]("https://powershell.myget.org/F/powershell-core-daily") is [6.1-preview831](https://ci.appveyor.com/project/PowerShell/powershell-f975h/build/v6.1.0-rc.831) or higher

Actual behavior
---------------

Downloaded PowerShell version from [MyGet]("https://powershell.myget.org/F/powershell-core-daily") is stuck at 6.1.0-preview-761

Environment data
----------------

```powershell
> $PSVersionTable
Name                           Value
----                           -----
PSVersion                      6.1.0
PSEdition                      Core
GitCommitId                    6.1.0
OS                             Microsoft Windows 10.0.17134
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### PR DESCRIPTION
**Pull request by adityapatwardhan**
Fixes #7955 

# PR Summary

Refactor CI yaml to move xunit verify to a template. 

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [ ] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)

### CODE REVIEW

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-macos`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-linux`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-windows`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-windows`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-static-analysis`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-macos`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-linux`

File: .vsts-ci/windows-daily.yml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+      - win_build
+
+- template: templates/windows-test.yml
+  parameters:
+    purpose: ElevatedPesterTests
+    tagSet: Others
+    parentJobs:
+      - win_build
+
+- template: templates/windows-packaging.yml
```
Comment (TravisEz13) [inline .vsts-ci/windows-daily.yml]: Add a comment why we wait here and in the CI yml why we don't

Comment (adityapatwardhan) [reply .vsts-ci/windows-daily.yml]: Fixed.

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-linux`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-macos`

**Comment by PoshChan** [comment]
@adityapatwardhan, your last commit had failures in `PowerShell-CI-windows`

**Review by TravisEz13 — APPROVED** [review]

**Pull request merged**